js/test_util.ts

```diff
@@ -49,6 +49,26 @@ export function test(fn: testing.TestFunction) {
   testPerm({ write: false, net: false, env: false }, fn);
 }
 
+interface Deferred {
+  promise: Promise<void>;
+  resolve: Function;
+  reject: Function;
+}
+
+export function deferred(): Deferred {
+  let resolve: Function | undefined;
+  let reject: Function | undefined;
+  const promise = new Promise<void>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return {
+    promise,
+    resolve: resolve!,
+    reject: reject!
+  };
+}
```

Comment:
Move this to `js/util.ts` as it's a general purpose utility.

Reply:
Done